src/components/scripts/oauth.mjs

```diff
@@ -0,0 +1,221 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export class OAuthConnector {
+  #redirectUri;
+  #clientId;
+  #clientSecret;
+  #oauthScopes = [];
+  #socketServer;
+  #authenticationCode;
+  #accessCode;
+  #refreshCode;
+  #promiseResolver;
+  #promiseRejecter;
+
+  #authentication_endpoint;
+  #token_endpoint;
+
+  constructor(authentication_endpoint, token_endpoint) {
+    this.#authentication_endpoint = authentication_endpoint;
+    this.#token_endpoint = token_endpoint;
+
+    this.#socketServer = new TCPServerSocket("::");
+  }
+
+  getOAuthAccessCode(client_id, client_secret, scopes) {
+    this.#clientId = client_id;
+    this.#clientSecret = client_secret;
+    this.#oauthScopes = scopes;
+
+    const { promise, resolve, reject } = Promise.withResolvers();
+    this.#promiseResolver = resolve;
+    this.#promiseRejecter = reject;
+
+    this.#socketServer.opened.then(
+      (server) => {
+        this.#redirectUri = `http://localhost:${server.localPort}`;
+        this.#runHttpServer();
+        this.#authenticate(this.#authentication_endpoint);
+      },
+    );
+
+    return promise;
+  }
+
+  get authentication_token() {
+    return this.#authenticationCode;
+  }
+
+  get access_token() {
+    return this.#accessCode;
+  }
+
+  get refresh_token() {
+    return this.#refreshCode;
+  }
+
+  #authenticate(endpoint) {
+    // Create <form> element to submit parameters to OAuth 2.0 endpoint.
+    let form = document.createElement("form");
```

Comment:
```suggestion
    const form = document.createElement("form");
```
Because we're not reassigning the variable (even though we're changing properties on it) it should be a `const`. Same applies elsewhere for this suggestion

Reply:
Done.